cpu/atmega2560/periph/gpio.c

```diff
@@ -102,6 +102,24 @@ int gpio_init(gpio_t pin, gpio_dir_t dir, gpio_pp_t pullup)
     else {
         _SFR_MEM8(_ddr_addr(pin)) &= ~(1 << _pin_num(pin));
         res = bit_is_clear(_SFR_MEM8(_ddr_addr(pin)), _pin_num(pin));
+
+        if (res == 0) {
```

Comment:
@danielamkaer The problem was the wrong macro in the lower lines, at the time I tested it I tried something like `init_in PORTA 4 1` and it had returned -1 :)

But now it should work :)